lib/fluent/plugin/out_honeycomb.rb

```diff
@@ -63,9 +64,17 @@ def write(chunk)
           log.debug "Skipping record #{record}"
           next
         end
-        if @sample_rate > 1 && rand(1..@sample_rate) > 1
-          next
+
+        if @presampled_key && record.include?(@presampled_key)
+          sample_rate = record.delete(@presampled_key)
```

Comment:
We might want to check the type and sign of the sample rate value we pull out of the record here. If we've configured, say, `presampled_key=sample_rate`, but then accidentally pass events that have `{"sample_rate": "some string value", ...}` or `{"sample_rate": -22, ...}`, those'll be rejected by the API, potentially leading to hard-to-debug failures.

Reply:
My question would be what should the plugin do if such validation failed with the message?

Reply:
Presumably, based on the current implementation of the plugin would result in the events being attempted to be sent `retry_limit` times (which appears to default to 17). It seems better to handle known bad data when parsing the record, over retrying each failed record because the sample rate is bad.

What should the action be when the `presampled_key` is bad? My gut instinct is `log.warn` plus disregard the `presampled_key` and pass the record on (ignoring any `sample_rate` set in the configuration, because the intended action seems like the user would want that data sent to honeycomb, even if not properly weighted).

Reply:
^ completely agree.

Reply:
PR updated with validation

Reply:
👍